chia/cmds/show.py

```diff
@@ -57,236 +89,305 @@ async def print_connections(client, time, NodeType, trusted_peers: Dict):
         print(con_str)
 
 
+async def execute_with_node(rpc_port: Optional[int], function: Callable, *args):
+    import traceback
+
+    import aiohttp
+
+    from chia.util.config import load_config
+    from chia.util.default_root import DEFAULT_ROOT_PATH
+    from chia.util.ints import uint16
+    from chia.rpc.full_node_rpc_client import FullNodeRpcClient
+
+    try:
+        config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+        self_hostname = config["self_hostname"]
+        if rpc_port is None:
+            rpc_port = config["full_node"]["rpc_port"]
+        node_client: FullNodeRpcClient = await FullNodeRpcClient.create(
+            self_hostname, uint16(rpc_port), DEFAULT_ROOT_PATH, config
+        )
+        await function(node_client, config, *args)
+
+    except Exception as e:
+        if isinstance(e, aiohttp.ClientConnectorError):
+            print(f"Connection error. Check if full node rpc is running at {rpc_port}")
+            print("This is normal if full node is still starting up")
+        else:
+            tb = traceback.format_exc()
+            print(f"Exception from 'show' {tb}")
+
+    node_client.close()
+    await node_client.await_closed()
+
+
 async def show_async(
```

Comment:
This method is getting very long and doing various things. This should be decomposed into smaller methods, one for each command.